app/src/main/java/org/systers/mentorship/view/activities/LoginActivity.kt

```diff
@@ -2,131 +2,234 @@ package org.systers.mentorship.view.activities
-        }
-
-        override fun onTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {
-        }
+// Composable for the View
+@Composable
+fun LoginScreen(viewModel: LoginViewModel, onSignUpButtonClick: () -> Unit) {
+
+    val username by viewModel.username.observeAsState("")
+    val password by viewModel.password.observeAsState("")
+    val buttonEnabled by viewModel.buttonEnabled.observeAsState(false)
+    val onUsernameChange: (String) -> Unit = { viewModel.onUsernameChange(it) }
+    val onPasswordChange: (String) -> Unit = { viewModel.onPasswordChange(it) }
+    val onLoginButtonClick: () -> Unit = { viewModel.onButtonClick() }
+    val scaffoldState = rememberScaffoldState()
+    val message by viewModel.message.observeAsState()
+    Scaffold(scaffoldState = scaffoldState, snackbarHost = { scaffoldState.snackbarHostState }) {
+        LoginContent(
+            username = username,
+            password = password,
+            onUsernameChange = onUsernameChange,
+            onPasswordChange = onPasswordChange,
+            isButtonEnabled = buttonEnabled,
+            onLoginButtonClick = onLoginButtonClick,
+            onSignUpButtonClick = onSignUpButtonClick,
+            scaffoldState = scaffoldState,
+            snackbarMessage = message
+        )
     }
+}
 
-    private fun validateCredentials(): Boolean {
-        var validCredentials = true
-        if (username.isBlank()) {
-            tiUsername.error = getString(R.string.error_empty_username)
-            validCredentials = false
-        } else {
-            tiUsername.error = null
-        }
-        if (password.isBlank()) {
-            tiPassword.error = getString(R.string.error_empty_password)
-            validCredentials = false
-        } else {
-            tiPassword.error = null
-        }
-        return validCredentials
-    }
+@Composable
+fun LoginContent(
+    username: String,
+    password: String,
+    onUsernameChange: (String) -> Unit,
+    onPasswordChange: (String) -> Unit,
+    isButtonEnabled: Boolean,
+    onLoginButtonClick: () -> Unit,
+    onSignUpButtonClick: () -> Unit,
+    scaffoldState: ScaffoldState,
+    snackbarMessage: String?
+) {
+    // Can be added as params to the function for making testing easier
+    val scope = rememberCoroutineScope()
+    val focusManager = LocalFocusManager.current
+    val signUpFocusRequest = remember { FocusRequester() }
+    val passwordFocusRequest = remember { FocusRequester() }
+
+    Box {
+        Column(
+            Modifier
+                .fillMaxSize()
+                .verticalScroll(rememberScrollState()),
+            verticalArrangement = Arrangement.SpaceAround,
+            horizontalAlignment = Alignment.CenterHorizontally
+        ) {
+
+            // This LaunchedEffect will display a snackbar each time the snackbar message changes
+            LaunchedEffect(key1 = snackbarMessage) {
+                scope.launch {
+                    snackbarMessage?.let {
```

Comment:
use takeif instead of/before let and check if it is not empty there

Reply:
Thanks I had no idea that existed.